### PR TITLE
backporting-docs: Add auto-backport

### LINKED ISSUE #286: Update Backport Documentation
With the [auto-backport](https://github.com/kata-containers/kata-containers/pull/4775) action, auto-backporting has been added. I want to update the documentation to reflect this.

### PR DESCRIPTION
**Pull request by bookinabox**
Updates auto-backport documentation and changes instances of master with
main.

Fixes #286

Signed-off-by: Derek Lee <derlee@redhat.com>

### CODE REVIEW

File: Backport-Guide.md [hunk 1]
```diff
@@ -77,19 +77,41 @@ After the backporting:
...
+A---B---C---D---E---F---G (main branch)
 ```
 
 After the backport, the `stable-1.2` branch contains commits  `A`, `B` and
 `C`, `E` and `G`.
 
 ## Backport Workflow
 
+### Auto-backporting
+Auto-backporting is a helpful feature than can be used to speed up the backport process. The project uses [Backport Action](https://github.com/marketplace/actions/backport-action) to handle the backporting and another GitHub workflow to automatically add the `auto-backport` label.
```
Comment (jodh-intel) [inline Backport-Guide.md:89]: Nit: overly long line. These make diffs bigger and harder to review.

File: Backport-Guide.md [hunk 2]
```diff
@@ -77,19 +77,41 @@ After the backporting:
...
+
+The following PRs will automatically have the `auto-backport` label added:
+- Corresponding issue contains the `bug` label
```
Comment (jodh-intel) [inline Backport-Guide.md:92]:
```suggestion
- PRs whose [associated issue](CONTRIBUTING.md#submit-issues-before-prs) has the `bug` label set.
```

This is still a bit unwieldy, but

File: Backport-Guide.md [hunk 3]
```diff
@@ -77,19 +77,41 @@ After the backporting:
...
+- PR contains `backport-needed` label
+
+PRs with the `no-backport-needed` label will automatically have its `auto-backport` label removed, superseding the above conditions.
```
Comment (jodh-intel) [inline Backport-Guide.md:95]:
```suggestion
PRs with the `no-backport-needed` label will automatically have their `auto-backport` label removed, superseding the above conditions.
```

File: Backport-Guide.md [hunk 4]
```diff
@@ -77,19 +77,41 @@ After the backporting:
...
+
+Additionally, the `auto-backport` label can be added or removed manually.
+The check for labels is triggered whenever a label is added or removed from the PR.
```
Comment (jodh-intel) [inline Backport-Guide.md:98]: You could make this line a "[note](https://github.com/kata-containers/kata-containers/blob/main/docs/Documentation-Requirements.md#notes)]":

```suggestion
> **Note:** The check for labels is triggered whenever a label is added or removed from the PR.
```

File: Backport-Guide.md [hunk 5]
```diff
@@ -77,19 +77,41 @@ After the backporting:
...
+
+The Backport Action executes the following steps
+- `auto-backport` label is checked when the PR is merged
```
Comment (jodh-intel) [inline Backport-Guide.md:101]: Missing period.

File: Backport-Guide.md [hunk 6]
```diff
@@ -77,19 +77,41 @@ After the backporting:
...
+- The action will look for all labels of the form`backport-to-BRANCHNAME` .
```
Comment (jodh-intel) [inline Backport-Guide.md:102]:
```suggestion
- The action will look for all labels of the form`backport-to-BRANCHNAME`.
```

File: Backport-Guide.md [hunk 7]
```diff
@@ -77,19 +77,41 @@ After the backporting:
...
+- The action will then attempt to generate a backport for each label, leaving a notification reporting success or failure.
+
+This does require labels to be made and added to the corresponding branches that one intends to backport to.
```
Comment (jodh-intel) [inline Backport-Guide.md:105]: This feels like a "note" too.

**Review by jodh-intel — COMMENTED** [review]
Thanks @bookinabox - a few comments...

**Comment by bookinabox** [comment]
Thanks for the feedback!

**Review by wainersm — APPROVED** [review]

**Comment by wainersm** [comment]
hey @jodh-intel could you pls review again?

**Review by jodh-intel — APPROVED** [review]
Thanks @bookinabox.

**Pull request merged**